ref: Render OneLogin idp selector with crispy

evanpurkhiser opened the pull request.

evanpurkhiser committed:
typo

evanpurkhiser committed:
ref: Render OneLogin idp selector with crispy

mattrobenolt commented on src/sentry_auth_saml2/onelogin/templates/sentry_auth_onelogin/select-idp.html:
```diff
@@ -15,37 +15,9 @@
-  </div>
+<form method="post" class="form-stacked">
+  {% include "sentry/partial/form_base.html" %}
+  <input type="hidden" name="provider" value="okta" />
```
Shouldn't this be `plugin.slug` ?

evanpurkhiser replied:
oh, the value is wrong too. Yeah it might as well just be that.

evanpurkhiser replied:
Actually sorry no, I don't think so. Unless I'm missing something we don't pass down the 'plugin' when rendering this, and I'm not even sure what that hidden input was doing before.

evanpurkhiser committed:
Correct provider hidden form value

evanpurkhiser commented:
whoops, that typo commit was a bit more than what should be in there  :)

evanpurkhiser committed:
Move processors into the forms classes

This allows us to subclass the forms and the process_metadata will still
know which processor to use for the form.

Slightly less clean than the mapping, since this couples the forms a bit
closer to the processors, but probably worth it.
